go/vt/vttablet/tabletmanager/action_agent.go

```diff
@@ -849,3 +870,12 @@ func (agent *ActionAgent) withRetry(ctx context.Context, description string, wor
 		}
 	}
 }
+
+func validateDemoteMasterType(tabletTypeParam string) (*topodatapb.TabletType, error) {
```

Comment:
We treat `topodatapb.TabletType` as a scalar type, so we usually pass it by value:

```suggestion
func validateDemoteMasterType(tabletTypeParam string) (topodatapb.TabletType, error) {
```

Reply:
Done!